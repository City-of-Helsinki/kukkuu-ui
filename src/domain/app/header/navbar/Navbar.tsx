import React, { FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';
import joinClassNames from 'classnames';

import styles from './navbar.module.scss';
import UserDropdown from '../userDropdown/UserDropdown';
import LanguageBar from './languageBar/LanguageBar';
import SmallScreenNav from './smallScreenNav/SmallScreenNav';
import { getCurrentLanguage } from '../../../../common/translation/TranslationUtils';

const Navbar: FunctionComponent = (props) => {
  const { t, i18n } = useTranslation();
  const history = useHistory();

  const currentLocale = getCurrentLanguage(i18n);
  const logoLang = currentLocale === 'sv' ? styles.sv : styles.fi;

  const isSmallScreen = window.innerWidth <= 768;
  return (
    <div className={styles.navbarTop}>
      <div className={styles.logoWrapper}>
        <div
          className={joinClassNames(styles.logo, logoLang)}
          onClick={() => history.push('/')}
        ></div>
        {!isSmallScreen && <div className={styles.appName}>{t('appName')}</div>}
      </div>
      <div className={styles.languageWrapper}>
        {isSmallScreen ? <SmallScreenNav /> : <LanguageBar />}
        {!isSmallScreen && <UserDropdown isSmallScreen={isSmallScreen} />}
      </div>
    </div>
  );
};

export default Navbar;
