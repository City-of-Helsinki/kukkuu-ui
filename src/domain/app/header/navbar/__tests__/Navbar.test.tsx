import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Navbar from '../Navbar';

describe('Navbar', () => {
  test('render normally', () => {
    const wrapper = shallow(<Navbar />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
