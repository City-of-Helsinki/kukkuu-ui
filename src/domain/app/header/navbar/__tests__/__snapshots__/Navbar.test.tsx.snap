// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar render normally 1`] = `
<div
  className="navbarTop"
>
  <div
    className="logoWrapper"
  >
    <div
      className="logo"
      onClick={[Function]}
    />
    <div
      className="appName"
    >
      Kulttuurin kummilapset
    </div>
  </div>
  <div
    className="languageWrapper"
  >
    <LanguageBar />
    <UserDropdown
      isSmallScreen={false}
    />
  </div>
</div>
`;
